# Dataset Integrity
I observe the three datasets and compare the CIAA-CIAA datasets with each other.

###### written by Nils Dyck, 15.11.2022

In [16]:
import pandas as pd

sep_parameter = ';'
default = pd.read_csv('../data/cleaned/gruber-holzer-rauch_dataset.csv', sep=sep_parameter)
ciaa_ciaa = pd.read_csv('../data/cleaned/CIAA-CIAA.csv', sep=sep_parameter)
dcfs_dcfs = pd.read_csv('../data/cleaned/DCFS-DCFS.csv', sep=sep_parameter)
dlt_dlt = pd.read_csv('../data/cleaned/DLT-DLT.csv', sep=sep_parameter)

ciaa_ciaa.head()

Author                                              Title  \
0              Yuan Gao                              In Memoriam Sheng Yu.   
1           Kai Salomaa                              In Memoriam Sheng Yu.   
2  Janusz A. Brzozowski       In Search of Most Complex Regular Languages.   
3   Andrzej Ehrenfeucht  A Formal Framework for Processes Inspired by t...   
4    Grzegorz Rozenberg  A Formal Framework for Processes Inspired by t...   

            Co-Authors Pagination  Published      Venue  
0          Kai Salomaa        1-4       2012  CIAA 2012  
1             Yuan Gao        1-4       2012  CIAA 2012  
2                  NaN       5-24       2012  CIAA 2012  
3   Grzegorz Rozenberg      25-27       2012  CIAA 2012  
4  Andrzej Ehrenfeucht      25-27       2012  CIAA 2012

## Compare CIAA and Default

In [17]:
default_titles = default['Title'].to_list()
ciaa_titles = ciaa_ciaa['Title'].to_list()

default_titles = set(default_titles)
ciaa_titles = set(ciaa_titles)

In [18]:
print(len(default_titles))
print(len(ciaa_titles))

687
700


In [19]:
ciaa_entries = ciaa_ciaa[['Title', 'Published', 'Venue']]
ciaa_entries = ciaa_entries.drop_duplicates()
ciaa_entries_without2022 = ciaa_entries[ciaa_ciaa.Published != 2022]
ciaa_entries

/var/folders/68/zr6z7wd56pl2hspyj9v4d57m0000gn/T/ipykernel_50978/588445184.py:3: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  ciaa_entries_without2022 = ciaa_entries[ciaa_ciaa.Published != 2022]


Title  Published      Venue
0                                 In Memoriam Sheng Yu.       2012  CIAA 2012
2          In Search of Most Complex Regular Languages.       2012  CIAA 2012
3     A Formal Framework for Processes Inspired by t...       2012  CIAA 2012
5                  Adding Pebbles to Weighted Automata.       2012  CIAA 2012
7     Typed Linear Algebra for Weigthed (Probabilist...       2012  CIAA 2012
...                                                 ...        ...        ...
1565  The Constrained Longest Common Subsequence Pro...       2007  CIAA 2007
1569  Finite Automata Accepting Star-Connected Langu...       2007  CIAA 2007
1570  Efficiently Matching with Local Grammars Using...       2007  CIAA 2007
1571                  Significant Subpatterns Matching.       2007  CIAA 2007
1572  A New Method for Compiling Parallel Replacemen...       2007  CIAA 2007

[707 rows x 3 columns]

The following numbers were collected "by hand". As you can see they match the results from the default and the "ciaa-ciaa_dataset"-script.

| Year | Papers | Authors |
| --- | --- | --- |
| 2022 | 19
| 2021 | 15
| 2020 | -
| 2019 | 19
| 2018 | 27
| 2017 | 20
| 2016 | 27
| 2015 | 26
| 2014 | 25
| 2013 | 31
| 2012 | 33
| 2011 | 27
| 2010 | 34
| 2009 | 32
| 2008 | 30
| 2007 | 34
| 2006 | 32
| 2005 | 36
| 2004 | 41
| 2003 | 33
| 2002 | 33
| 2001 | 24
| 2000 | 35
| 1999 | 16
| 1998 | 22
| 1997 | 22
| 1996 | 14
| Sum | 707